kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js

```diff
@@ -0,0 +1,112 @@
+import angular from 'angular';
+
+import '../../services/panel';
+import '../../services/api/collections-api';
+import '../../directives/gr-auto-focus';
+
+import './gr-collections-panel.css!';
+import nodeTemplate from './gr-collections-panel-node.html!text';
+
+export var grCollectionsPanel = angular.module('grCollectionsPanel', [
+    'kahuna.services.panel']);
+
+grCollectionsPanel.controller('GrCollectionsPanelCtrl', [
+    'panelService',
+    '$rootScope',
+    'collections',
+    function (panelService, $rootScope, collections) {
+
+    const ctrl = this;
+    const panelName = 'gr-collections-panel';
+
+    ctrl.isVisible = false;
+
+    panelService.addPanel(panelName, false);
+    panelService.available(panelName, true);
+
+    $rootScope.$on(
+        `ui:panels:${panelName}:updated`,
+        () => {
+            ctrl.isVisible = panelService.isVisible(panelName);
+        }
+    );
+
+    collections.getCollections().then(collections => {
+        ctrl.collections = collections.data.children;
+    });
```

Comment:
Definitely - I think this service needs a lot of work - it's just a works-enough to show people. I will add some TODOs